File: src/p0001/java/solution.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[]{};
        HashMap<Integer, Integer> dict = new HashMap<>();
        for (int j = 0; j < nums.length; j++) {
            int y = nums[j], x = target - y;
            Integer i = dict.get(x);
            if (i != null) {
                result = new int[]{i, j};
                break;
            }
            dict.put(y, j);
        }
        return result;
    }
}

File: src/p0001/rust/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut result = Vec::new();
        let mut dict = HashMap::new();
        for (j, y) in nums.iter().enumerate() {
            let x = target - *y;
            if let Some(i) = dict.get(&x) {
                result.push(*i);
                result.push(j as i32);
                break;
            }
            dict.insert(*y, j as i32);
        }
        return result;
    }
}
